server/src/services/MailerService.ts

```diff
@@ -92,3 +92,23 @@ export default class MailerService {
     }
   }
 }
+
+export interface BatchEmailData {
+  email: string;
+  subject: string;
+  text: string;
+  options?: object;
+}
+
+export async function batchSender(
+  mailData: () => Generator<BatchEmailData, void>,
+) {
+  for (const { email, subject, text, options } of mailData()) {
+    await new MailerService({
+      emailList: [email],
+      subject,
+      htmlEmail: text,
+      ...options,
+    }).sendEmail();
```

Comment:
This waits until each email has been sent before starting to send the next one.  Is that the plan?  It's possible to wait for the whole batch, if that's desired.

Reply:
_Plan_ is a strong word, not really a plan. I didn't think too much about it, I might have assumed it's possible it will be used with potentially high number. Which approach do you think would be better?

Reply:
So, generally what I do if there's a bunch of async tasks I want to perform and then wait for them all to be done is the following.  Start by creating an array (of email data, in this case), map that to an array of Promises and then `await Promise.all(theArrayOfPromises)`.

That way all the async calls get fired as soon as possible and you still control the size of the batch.

Does that make sense? If not, I can show you some code.

Reply:
I feel like we already had similar conversation somewhere :stuck_out_tongue:. I've added change, see if that works for you.

Reply:
It does seem eerily familiar!

Yeah, this seems good, thanks :)